applications/concept-catalogue/src/main/kotlin/no/begrepskatalog/conceptregistration/BegreperApiImplK.kt

```diff
@@ -48,30 +51,79 @@ class BegreperApiImplK(val sqlStore: SqlStore) : BegreperApi {
                 }
     }
 
-    override fun setBegrepById(httpServletRequest: HttpServletRequest, @ApiParam(value = "id", required = true) id: String, @ApiParam(value = "", required = true) @Valid @RequestBody begrep: Begrep, @ApiParam(value = "Om begrepet skal valideres") @Valid @RequestParam(value = "validate", required = false) validate: Boolean): ResponseEntity<Begrep> {
-        //We must have an ID that already exists, AND a virksomhet
-        if (!sqlStore.begrepExists(begrep)) {
+    override fun setBegrepById(httpServletRequest: HttpServletRequest?, id: String?, begrep: Begrep?, validate: Boolean?): ResponseEntity<Begrep> {
+        if (id == null) {
+            throw RuntimeException("Attempt to PATCH begrep with no id path variable given")
+        }
+        if (begrep == null) {
+            throw RuntimeException("Attempt to PATCH begrep with no begrep data given. Id provided was $id")
+        }
+
+        if (!sqlStore.begrepExists(id)) {
             throw RuntimeException("Attempt to PUT begrep that does not already exist. Begrep id ${begrep.id}")
         }
+        //Get the begrep, and just update
+        var storedBegrep = sqlStore.getBegrepById(id)
 
-        if (begrep.ansvarligVirksomhet == null) {
-            throw RuntimeException("Attempt to PUT begrep with no accompanying Ansvarlig Virksomhet")
+        if (storedBegrep == null) {
+            throw java.lang.RuntimeException("Stored begrep with id $id was null. This should not happen")
         }
 
-        if (begrep.status == Status.UTKAST) {
-            sqlStore.saveBegrep(begrep)
+        val updatedBegrep = updateBegrep(begrep, storedBegrep)
+
+        if (updatedBegrep.status == Status.UTKAST) {
```

Comment:
Status.UTKAST er det generert direkte fra spec? Det står 'utkast' i lowercase i spec, vet ikke om det har noe å si for koden?

Reply:
public enum Status {
  
  UTKAST("utkast"),
  
  GODKJENT("godkjent"),
  
  PUBLISERT("publisert");

Dette er det som blir autogenerert, ja